keywords/SyncGateway.py

```diff
@@ -561,11 +578,16 @@ def upgrade_sync_gateways(self, cluster_config, sg_conf, sync_gateway_version, u
                 log_info("Keyerror in getting logging{}".format(ex.message))
                 playbook_vars["logging"] = '{} {},'.format(logging_config, "}")
 
-            if get_sg_use_views(cluster_config):
-                playbook_vars["sg_use_views"] = '"use_views": true,'
+            server_version = get_cbs_version(cluster_config)
+            cbs_version, cbs_build = version_and_build(server_version)
+            if cbs_version >= "5.5.0":
+                if get_sg_use_views(cluster_config):
+                    playbook_vars["sg_use_views"] = '"use_views": true,'
+                else:
+                    num_replicas = get_sg_replicas(cluster_config)
+                    playbook_vars["num_index_replicas"] = '"num_index_replicas": {},'.format(num_replicas)
             else:
-                num_replicas = get_sg_replicas(cluster_config)
-                playbook_vars["num_index_replicas"] = '"num_index_replicas": {},'.format(num_replicas)
+                playbook_vars["sg_use_views"] = '"use_views": true,'
```

Comment:
I'm not sure if SG versions <5.5 recognises "use_views" as property. Can you check with Ben or Sri, regarding same? As far as I remember, use_views was added as to 5.5 for backward compatibility.

Reply:
comments from Adam: 
SG 2.1 uses GSI by default, which requires CBS 5.5 or later.  The use_views flag was included in SG 2.1 to support running w/ older versions of CBS (by setting use_views=true)
full details here https://docs.couchbase.com/sync-gateway/2.1/compatibility-matrix.html